imap_processing/ialirt/tests/unit/test_decom_ialirt.py

```diff
@@ -0,0 +1,91 @@
+
+
+@pytest.fixture(scope="session")
+def packet_path():
+    """Returns the raw packet directory."""
+    return Path(sys.modules[__name__.split(
+        '.')[0]].__file__).parent / 'ialirt' / 'tests' / 'test_data' / 'l0' / \
+        'IALiRT Raw Packet Telemetry.txt'
+
+
+@pytest.fixture(scope="session")
+def xtce_ialirt_path():
+    """Returns the xtce auxiliary directory."""
+    return Path(sys.modules[__name__.split(
+        '.')[0]].__file__).parent / 'ialirt' / 'packet_definitions' \
+        / "ialirt.xml"
+
+
+@pytest.fixture(scope="session")
+def decom_data(packet_path, xtce_ialirt_path):
+    """Data for decom_ultra"""
+    data_packet_list = generate_xarray(packet_path, xtce_ialirt_path)
+    return data_packet_list
+
+
+@pytest.fixture(scope="session")
+def decom_test_data(packet_path, xtce_ialirt_path):
+    """Read test data from file"""
+    data_packet_list = decom_packets(packet_path, xtce_ialirt_path)
+    return data_packet_list
+
+
+def test_length(decom_test_data):
+    """Test if total packets in data file is correct"""
+    total_packets = 32
+    assert len(decom_test_data) == total_packets
+
+
+def test_enumerated(decom_test_data):
+    """Test if enumerated values derived correctly"""
+
+    for packet in decom_test_data:
+
+        assert packet.data["SC_SWAPI_STATUS"].derived_value == 'NOT_OPERATIONAL'
+        assert packet.data["SC_MAG_STATUS"].derived_value == 'NOT_OPERATIONAL'
+        assert packet.data["SC_HIT_STATUS"].derived_value == 'NOT_OPERATIONAL'
+        assert packet.data["SC_CODICE_STATUS"].derived_value == 'NOT_OPERATIONAL'
+        assert packet.data["SC_LO_STATUS"].derived_value == 'NOT_OPERATIONAL'
+        assert packet.data["SC_HI_45_STATUS"].derived_value == 'NOT_OPERATIONAL'
+        assert packet.data["SC_HI_90_STATUS"].derived_value == 'NOT_OPERATIONAL'
+        assert packet.data["SC_ULTRA_45_STATUS"].derived_value == 'NOT_OPERATIONAL'
+        assert packet.data["SC_ULTRA_90_STATUS"].derived_value == 'NOT_OPERATIONAL'
+        assert packet.data["SC_SWE_STATUS"].derived_value == 'NOT_OPERATIONAL'
+        assert packet.data["SC_IDEX_STATUS"].derived_value == 'NOT_OPERATIONAL'
+        assert packet.data["SC_GLOWS_STATUS"].derived_value == 'NOT_OPERATIONAL'
+        assert packet.data["SC_SPINPERIODVALID"].derived_value == 'INVALID'
+        assert packet.data["SC_SPINPHASEVALID"].derived_value == 'INVALID'
+        assert packet.data["SC_ATTITUDE"].derived_value == 'SUNSENSOR'
+        assert packet.data["SC_CATBEDHEATERFLAG"].derived_value == 'ON'
+        assert packet.data["SC_AUTONOMY"].derived_value == 'OPERATIONAL'
+        assert packet.data["HIT_STATUS"].derived_value == 'OFF-NOMINAL'
+        assert packet.data["SWE_NOM_FLAG"].derived_value == 'OFF-NOMINAL'
+        assert packet.data["SWE_OPS_FLAG"].derived_value == 'NON-HVSCI'
+
+
+def test_decom_instruments(decom_data, decom_test_data):
+    """This function checks that all instrument parameters are accounted for."""
```

Comment:
I'm not sure I fully follow all of this test. I think it'd help to break this down even further and actually assert on individual instruments, possibly parameterizing this over instruments if that would help.

But, I think it'd be good to say some things about each instrument individually and spell it out explicitly here. `assert len(decom_data["HIT"]["data"]) == expected HIT data size`

or, I don't think it would be unreasonable to also make an xarray dataset manually with `np.zeros(proper_shape)` and set the values you expect here so that you are asserting against a known zero array of the proper shape without having to decipher the loops within the test.